### PR TITLE
Digital ocean defect for inventory

### PR DESCRIPTION
**Pull request by rahulray19**
Changes done for digital ocean defects-
1-provisioned VMs missing Internal/External IP
2-Inventory: IP Address (appears on first sync, cleared on subsequent syncs)
3-Inventory: General Operating System - linux/windows (if possible)
4-Inventory: Total Storage (if possible)

### CODE REVIEW

File: src/main/groovy/com/morpheusdata/digitalocean/cloud/sync/VirtualMachineSync.groovy [hunk 1]
```diff
@@ -259,7 +275,7 @@ class VirtualMachineSync {
                 maxMemory        : cloudItem.memory * ComputeUtility.ONE_MEGABYTE,
                 maxCores         : (cloudItem.vcpus?.toLong() ?: 0) * (cloudItem.disk?.toLong() ?: 0),
                 coresPerSocket   : 1l,
-                osType           : 'unknown',
+                osType           : cloudItem.image ? cloudItem.image?.distribution : 'unknown',
```
Comment (ddeyoung) [inline src/main/groovy/com/morpheusdata/digitalocean/cloud/sync/VirtualMachineSync.groovy]: osType should just be "linux" or "windows":
```
def osType = cloudItem.image?.distribution?.toLowerCase()?.contains('windows') ? 'windows' : 'linux'
...
[
    osType: osType
]
```

File: src/main/groovy/com/morpheusdata/digitalocean/cloud/sync/VirtualMachineSync.groovy [hunk 2]
```diff
@@ -259,7 +275,7 @@ class VirtualMachineSync {
...
                 osDevice         : '/dev/vda',
                 serverOs         : new OsType(code: 'unknown'),
```
Comment (ddeyoung) [inline src/main/groovy/com/morpheusdata/digitalocean/cloud/sync/VirtualMachineSync.groovy]: We can make a decent effort to find an OS match:
```
private Map<String, OsType> getAllOsTypes() {
  osTypes ?: (osTypes = morpheusContext.async.osType.listAll().toMap {it.name.toLowerCase()}.blockingGet())
}

def findOsMatch(distribution, osType) {
    allOsTypes[disribution.toLowerCase() ?: allOsTypes[osType] ?: new OsType(code:'unknown')
}

[
  ...
  serverOs: findOsMatch(osType)
  ...
]
```

**Review by ddeyoung — CHANGES_REQUESTED** [review]

File: src/main/groovy/com/morpheusdata/digitalocean/cloud/sync/VirtualMachineSync.groovy (hunk 1 above)
Comment (rahulray19) [reply src/main/groovy/com/morpheusdata/digitalocean/cloud/sync/VirtualMachineSync.groovy]: @ddeyoung changes done

File: src/main/groovy/com/morpheusdata/digitalocean/cloud/sync/VirtualMachineSync.groovy (hunk 2 above)
Comment (rahulray19) [reply src/main/groovy/com/morpheusdata/digitalocean/cloud/sync/VirtualMachineSync.groovy]: @ddeyoung changes done

**Review by ddeyoung — APPROVED** [review]

**Pull request merged**